### PR TITLE
Created release script that updates all ckeditor5 dependencies to the latest version.

### PR DESCRIPTION
**Pull request by przemyslaw-zan**
Internal: Created release script that updates all \`ckeditor5\` dependencies to the latest version. Closes cksource/ckeditor5-internal#1123.

### CODE REVIEW

File: scripts/release/bump-versions.js [hunk 1]
```diff
@@ -17,7 +17,7 @@
 // You can test the whole process using `dry-run` mode. It won't change anything in the project
 // and any repository.
 //
-// This task must be called before: `npm run release:publish`.
+// This task must be called before: `npm run release:update-package-versions` and npm `run release:publish`.
```
Comment (psmyrek) [inline scripts/release/bump-versions.js:20]:
```suggestion
// This task must be called before: `npm run release:update-package-versions` and `npm run release:publish`.
```

**Review by psmyrek — CHANGES_REQUESTED** [review]

**Review by psmyrek — APPROVED** [review]
LGTM 🎉 

It should be merged after the `ckeditor5-dev-env` is released and contains changes from https://github.com/ckeditor/ckeditor5-dev/pull/737.

**Review by psmyrek — APPROVED** [review]
LGTM 🎉 

Found just minor cosmetic change to consider.

File: scripts/release/update-package-versions.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+// See https://github.com/cksource/ckeditor5-internal/issues/1123.
+//
+// This task must be called before: `npm run release:publish`.
+//
+// Use:
+// npm run release:update-package-versions -- --dry-run --diff
+//
+// Available arguments:
+// --dry-run:   Prevents the script from committing, and instead shows list of files that were changed in `packages/*` directory.
+// --diff:      Instead of list of files, it displays detailed list of changes from each file. Has no effect without --dry-run.
```
Comment (pomek) [inline scripts/release/update-package-versions.js:24]: Please, merge these options. While the `--dry-run` option makes sense and allows testing what the script will do, I don't understand how it should work when executing `npm run release:update-package-versions -- --diff`.

File: scripts/release/update-package-versions.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+//
+// Use:
+// npm run release:update-package-versions -- --dry-run
+//
+// Available arguments:
+// --dry-run:   Prevents the script from committing, and instead displays detailed list of changes from each file that was changed.
+
+const CKEDITOR5_PATH = path.resolve( __dirname, '..', '..' );
+const CKEDITOR5_INTERNAL_PATH = path.resolve( __dirname, '..', '..', 'external', 'ckeditor5-internal' );
+const COLLABORATION_FEATURES_PATH = path.resolve( __dirname, '..', '..', 'external', 'collaboration-features' );
```
Comment (psmyrek) [inline scripts/release/update-package-versions.js:30]: Minor idea to consider: paths to `ckeditor5-internal` and `collaboration-features` could be now based on the `CKEDITOR5_PATH` constant, so when the location of this script changes in the future, only one path to the CKEditor 5 will have to be adapted.

**Review by psmyrek — CHANGES_REQUESTED** [review]

File: packages/ckeditor5-adapter-ckfinder/package.json [hunk 4]
```diff
@@ -18,7 +18,7 @@
     "@ckeditor/ckeditor5-basic-styles": "^31.1.0",
     "@ckeditor/ckeditor5-core": "^31.1.0",
     "@ckeditor/ckeditor5-clipboard": "^31.1.0",
-    "@ckeditor/ckeditor5-dev-utils": "^27.1.0",
+    "@ckeditor/ckeditor5-dev-utils": "27.1.0",
```
[the remark this answers is not in the document]
Comment (przemyslaw-zan) [reply packages/ckeditor5-adapter-ckfinder/package.json:21]: Oops, pushed the script-generated commit as a mistake. Will revert this in a second. Why did it changed though... I'll need to investigate whether or not this is a bug in my code.

File: packages/ckeditor5-adapter-ckfinder/package.json [hunk 5]
```diff
@@ -29,8 +29,8 @@
     "@ckeditor/ckeditor5-typing": "^31.1.0",
     "@ckeditor/ckeditor5-undo": "^31.1.0",
     "@ckeditor/ckeditor5-upload": "^31.1.0",
-    "webpack": "^5.58.1",
-    "webpack-cli": "^4.9.0"
+    "webpack": "5.58.1",
+    "webpack-cli": "4.9.0"
```
[the remark this answers is not in the document]
Comment (przemyslaw-zan) [reply packages/ckeditor5-adapter-ckfinder/package.json:33]: Same as https://github.com/ckeditor/ckeditor5/pull/11007#discussion_r772917649

File: scripts/release/update-package-versions.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+	.updatePackageVersions( {
+		cwd: process.cwd(),
+		dryRun: process.argv.includes( '--dry-run' ),
+		pathsToUpdate: [
+			{ path: CKEDITOR5_PATH + '/packages', commit: true },
+			{ path: CKEDITOR5_PATH + '/release', commit: false },
+			{ path: CKEDITOR5_INTERNAL_PATH + '/packages', commit: true },
+			{ path: CKEDITOR5_INTERNAL_PATH + '/release', commit: false },
+			{ path: COLLABORATION_FEATURES_PATH + '/packages', commit: true },
+			{ path: COLLABORATION_FEATURES_PATH + '/release', commit: false }
```
[the remark this answers is not in the document]
Comment (przemyslaw-zan) [reply scripts/release/update-package-versions.js:50]: I can update this. But I've included `directoryPath.split( '\u005C' ).join( '\u002F' )` in the script itself. Should I remove that safety step, and trust that all arguments passed in the future will be Windows-proof?

[the remark this answers is not in the document]
Comment (pomek) [reply scripts/release/update-package-versions.js:58]: The script should change only the CKEditor 5 project. Other repositories follow their release steps, and we need to update them. But AFAIR, we wanted to touch only the internal repository, though.

[the remark this answers is not in the document]
Comment (przemyslaw-zan) [reply scripts/release/update-package-versions.js:58]: So... `ckeditor5/packages` and `ckeditor5/release` only?

[the remark this answers is not in the document]
Comment (pomek) [reply scripts/release/update-package-versions.js:58]: In `ckeditor5` all updates will be handled by existing release scripts. Hence, we should move this script to the internal repository.

**Comment by psmyrek** [comment]
Closing because of https://github.com/ckeditor/ckeditor5/pull/11007#discussion_r773278443.

**Pull request closed without merging**